### ISSUE TITLE
Old dependencies

### ISSUE THREAD

**Issue by jayvdb** [maxogden/ndjson#18]
This package is using some quite old versions of dependencies, such as split2 0.2.1 when the latest is 2.1.1.
In the case of split 0.2.1, this is likely the cause of some junk in our docker image.
See https://github.com/coala/docker-coala-base/issues/120

**Comment by jayvdb** [comment]
@mcollina can you confirm that the split2 dependency can be upgraded from 0.2.1 to 2.1.1 painlessly.

**Comment by mcollina** [comment]
From 0.2.1 to 2.0.1 is a huge step, but it should be painless.
Several things are different between those two versions, so you'd want to bump he major of ndjson as well.

**Comment by jayvdb** [comment]
@mcollina, are all the options for `parse` in 0.2.1 still supported?
That appears to be the only usage in this package:
https://github.com/maxogden/ndjson/blob/2659c24280d45aaf38b33bf7beed2ea34f424e48/index.js#L24

**Comment by mcollina** [comment]
Yes it's fine. v2 just handles correctly utf8 chars, plus it's an updated dependency on readable-stream.

**Comment by jayvdb** [comment]
So why wouldnt it be backwards compatible, or...
why would it need to be a major release of ndjson?

**Comment by mcollina** [comment]
because I consider a substantial change of the implementation a major change, ad it might introduce bugs. 

It's up to @maxogden and @finnp.

**Comment by millette** [comment]
If it follows [semantic versioning](http://semver.org/), quoting:

> Given a version number MAJOR.MINOR.PATCH, increment the:
> 
> 1. MAJOR version when you make incompatible API changes,
> 1. MINOR version when you add functionality in a backwards-compatible manner, and
> 1. PATCH version when you make backwards-compatible bug fixes.

It's all about compatibility, nothing about the implementation itself. If you don't change the interface (including supported node versions), I suggest you only make it a PATCH change.

**Comment by mcollina** [comment]
As the maintainer of both split2 and readable-stream, I still recommend a major bump. 
Specifically, split2 handles correctly utf8 chars, and that alone can cause applications using this to behave differently: they can break, ence the major bump. So yes, it’s about compatibility.

**Comment by millette** [comment]
Good point @mcollina  :-)

Sorry if that came out pedantic,
